libbeat/processors/syslog/syslog.go

```diff
@@ -0,0 +1,207 @@
+type processor struct {
+	config
+
+	log   *logp.Logger
+	stats processorStats
+}
+
+// processorStats contains the metrics fields for the syslog processor.
+type processorStats struct {
+	// Success measures the number of successfully parsed syslog messages.
+	Success *monitoring.Int
+	// Failure measures the number of occurrences where a message was unable to be parsed.
+	Failure *monitoring.Int
+	// Missing measures the number of occurrences where an event was missing the required input field.
+	Missing *monitoring.Int
+}
+
+// init will register various aspects of this processor.
+func init() {
+	processors.RegisterPlugin(procName,
+		checks.ConfigChecked(New,
+			checks.RequireFields(
+				"field",
+			),
+			checks.AllowedFields(
+				"field",
+				"format",
+				"timezone",
+				"overwrite_keys",
+				"ignore_missing",
+				"ignore_failure",
+				"tag",
+			),
+		),
+	)
+	jsprocessor.RegisterPlugin("Syslog", New)
+}
+
+// defaultConfig will return a config with default values.
+func defaultConfig() config {
+	return config{
+		Field:         "message",
+		Format:        syslog.FormatAuto,
+		TimeZone:      cfgtype.MustNewTimezone("Local"),
+		OverwriteKeys: true,
+	}
+}
+
+// New creates a new processor from the provided configuration, or an error if the configuration is invalid.
+func New(c *common.Config) (processors.Processor, error) {
+	cfg := defaultConfig()
+
+	if err := c.Unpack(&cfg); err != nil {
+		return nil, fmt.Errorf("fail to unpack the "+procName+" processor configuration: %w", err)
+	}
+
+	id := int(instanceID.Inc())
+	log := logp.NewLogger(logName).With("instance_id", id)
+	registryName := logName + "." + strconv.Itoa(id)
+
+	if cfg.Tag != "" {
+		log = log.With("tag", cfg.Tag)
+		registryName = logName + "." + cfg.Tag + "-" + strconv.Itoa(id)
+	}
+	registry := monitoring.Default.NewRegistry(registryName, monitoring.DoNotReport)
+
+	return &processor{
+		config: cfg,
+		log:    log,
+		stats: processorStats{
+			Success: monitoring.NewInt(registry, "success"),
+			Failure: monitoring.NewInt(registry, "failure"),
+			Missing: monitoring.NewInt(registry, "missing"),
+		},
+	}, nil
+}
+
+// Run will process an event and will update the fields based on the parsed message, or an error if the
+// message could not be parsed. If an error occurs and the configuration is set to not ignore errors,
+// the 'error.message' field will be set with error that was encountered.
+func (p *processor) Run(event *beat.Event) (*beat.Event, error) {
+	if err := p.run(event); err != nil && !p.IgnoreFailure {
+		err = fmt.Errorf(procName+" failed to process field %q: %w", p.Field, err)
+		appendStringField(event.Fields, "error.message", err.Error())
+		return event, err
+	}
+
+	return event, nil
+}
+
+// run will parse the event and populate fields on the event.
+func (p *processor) run(event *beat.Event) error {
+	value, err := event.GetValue(p.Field)
+	if err != nil {
+		if err == common.ErrKeyNotFound {
```

Comment:
For future compatibility with wrapped errors: `errors.Is(err, common.ErrKeyNotFound)`